### ISSUE TITLE
Improve Analyze.locate_signficant_halite()

### ISSUE THREAD

**Issue by dgets** [dgets/d4m0Turtle#20]
It has occurred to me that if `locate_significant_halite()` were to be modified a bit, it may save quite a bit of trouble on the front of `minimum_distance_processing()` (see also issue #11).  At this point, the sole usage for that method is to obtain a little bit of free space around the shipyard, so that dropoffs, and spawning new ships, will be possible.  This need could be eradicated, for the most part, if the `PerimeterSearch` methods could be modified to allow seeking out of halite between a perimeter _and a minimum distance_, instead of just within the perimeter in general.

Of course this would require #2 having debugging completed.  The current algorithm is still buggy.

**Comment by emilmirzayev** [comment]
One good approach could be dividing map into chunks and have all meta info in some Brain class.
For ships, make also a local brain for keeping info only for the chunk they are in.

**Comment by dgets** [comment]
@emilmirzayev That's true.  I had been thinking about the `Perimeter` style search primarily for ease of ranking search results by distance.  Now that I think about it, though, with the `hlt.GameMap.calculate_distance()` functionality, it may be nearly as efficient to just quickly rank the results of the surrounding map chunks and go from there.  Thank you for the thought!

**Comment by emilmirzayev** [comment]
We can team up on this challenge if you are up to it!

**Comment by dgets** [comment]
@emilmirzayev: I'm up for it, didn't mean to be ignoring you, life has just been kicking my ass a bit.  However, I'm kind of thinking that, at this point, a rewrite may be in order, too.  I've realized a lot of things that I should've done differently...  Gimme a little bit to ponder that, either way I'm up for collaborating from here on out...

**Comment by emilmirzayev** [comment]
@dgets , shure man! Take your time. I'll be around :)

**Comment by dgets** [comment]
@emilmirzayev: K.  I started work on _nightMiner_ yesterday in order to detangle the cluster of features that I'd attempted [rather foolishly] to all add at once in _d4m0Turtle_.  It's coming along decently.  If you want you can look around a bit (warning, I did mess up the dropoff procedure in the branch I'm in recently; I think I may've accidentally merged that somewhere else, too-- I'll be checking that out soon here), or else I can just let you know when I get back to the stage where I'll be working on the `Brain` class you were talking about.  :)

**Comment by dgets** [comment]
@emilmirzayev: just wanted to let you know that I'm at a similar location, but in a somewhat better coded bot, at https://github.com/dgets/nightMiner right now.  If you wanted to take a look at things that I've already started, and see how they compare to what you were thinking of, the applicable code is in _analytics.py_'s `MapChunk` object.  I'd love to collaborate if you're still up for it, over there.

**Comment by emilmirzayev** [comment]
Hey there! 
Just got an email with your comment, I have missed the previous one you made 19 days ago. I will now take a look at the new code you have written.
I am about to have very major changes in my life these days, presumably the entire December. Yet, I had our talk in my mind the entire time.
Any way we can communicate outside GitHub issue commenting?
Like email, instant messaging (telegram maybe)?
best,

**Comment by dgets** [comment]
@emilmirzayev: Yeah something outside of github would be preferable for some things, wouldn't it?  You can hit me up at damo [dot] gets [at] gmail, if you like.  We don't have to be stuck to email for contact, we can use an instant messenger or whatever, but I'm not gonna put my contact information for that up here, so email would be a good next step.  :)

I know what you mean about the life changes and activity levels.  No worries at all about things, life definitely comes first, this second.  If you don't get time to do as much as you wanted, it's all good.  One of the primary reasons that I'm doing this is problems getting development time, myself.  The other one being to try to learn to integrate development a bit better with other people.

Anyway, I'll tell you more when I get an email from you.  :)  Hope to hear from you soon!
